tests/sub_port_interfaces/conftest.py

```diff
@@ -435,6 +440,56 @@ def apply_tunnel_table_to_dut(duthost, apply_route_config):
         duthost.command('docker exec -i database redis-cli -n 4 -c DEL "TUNNEL|MuxTunnel{}"'.format(index))
 
 
+@pytest.fixture()
+def apply_balancing_config(duthost, ptfhost, ptfadapter, define_sub_ports_configuration, apply_config_on_the_dut, apply_config_on_the_ptf):
+    """
+    Apply balancing configuration on the DUT and remove after tests
+    Args:
+        duthost: DUT host object
+        ptfhost: PTF host object
+        ptfadapter: PTF adapter
+        define_sub_ports_configuration: Dictonary of parameters for configuration DUT
+        apply_config_on_the_dut: fixture for applying sub-ports configuration on the DUT
+        apply_config_on_the_ptf: fixture for applying sub-ports configuration on the PTF
+    Yields:
+        Dictonary of parameters for configuration DUT and PTF host
+    """
+    new_sub_ports = []
+    sub_ports = define_sub_ports_configuration['sub_ports']
+    dut_ports = define_sub_ports_configuration['dut_ports']
+    ptf_ports = define_sub_ports_configuration['ptf_ports']
+
+    src_ports = tuple(set(ptfhost.host.options['variable_manager'].extra_vars.get("ifaces_map").values()).difference(ptf_ports))
+
+    network = u'1.1.1.0/24'
+    network = ipaddress.ip_network(network)
+
+    for port, subnet in zip(dut_ports.values(), network.subnets(new_prefix=30)):
+        sub_ports_on_port = [sub_port for sub_port in sub_ports if port + '.' in sub_port]
+
+        sub_port_neighbors = [sub_ports[sub_port]['neighbor_port'] for sub_port in sub_ports_on_port]
+        configure_ptf_nn_agent(ptfhost, ptfadapter, sub_port_neighbors)
```

Comment:
Can we use the vlan id to verify the load-balancing instead of changing the ptf port mapping dynamically?

Reply:
It's a great solution, but there may be problems with sorting packets from buffer per VLAN id. Sorting of packets will take much time and test time will be increased. Thanks!